nncf/torch/dynamic_graph/trace_tensor.py

```diff
@@ -50,36 +53,73 @@ def __str__(self):
         return "C{}_I{}_".format(self.creator_id, self.index) + "S" + "x".join([str(s) for s in self.shape])
 
 
-class TracedTensor(torch.Tensor):
+class TracedTensorMixin:
     """
-    When tracing a torch model, intermediate tensors will be dynamically turned into
-    instances of this class to be able to store additional data required for establishing
-    relation between tensor producer and consumer operations.
+    A mixin class providing tracing capabilities to PyTorch tensors.
+
+    This class provides interfaces for patching a given torch tensor and associating it with the provided tensor_meta.
     """
 
-    @staticmethod
-    def from_torch_tensor(tensor: torch.Tensor, tensor_meta: TensorMeta) -> "TracedTensor":
+    TENSOR_META = "tensor_meta"
+    ORIGINAL_CLASS = "original_class"
+    _TRACING_ATTRS = "_tracing_attrs"
+
+    @property
+    def tensor_meta(self):
+        return self.tracing_attrs[TracedTensorMixin.TENSOR_META]
+
+    @tensor_meta.setter
+    def tensor_meta(self, value: Union[None, TensorMeta]):
+        self.tracing_attrs[TracedTensorMixin.TENSOR_META] = value
+
+    @property
+    def tracing_attrs(self):
+        if not hasattr(self, TracedTensorMixin._TRACING_ATTRS):
+            self._tracing_attrs = {}
+        return self._tracing_attrs
+
+    @classmethod
+    def patch(cls: Type[_T], tensor: torch.Tensor, tensor_meta: Optional[TensorMeta] = None) -> _T:
```

Comment:
Type hint is not required for `cls` argument

for return type can be used classname in quotes or  `from __future__ import annotations` 
```python
-> "TracedTensorMixin":
```
or 
```python
from __future__ import annotations
...
-> TracedTensorMixin:
```

Reply:
done